packages/edit-post/src/components/options-modal/options/enable-custom-fields.js

```diff
@@ -1,47 +1,54 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
 import { withSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import BaseOption from './base';
 
-export class EnableCustomFieldsOption extends Component {
-	constructor( { isChecked } ) {
-		super( ...arguments );
-
-		this.toggleCustomFields = this.toggleCustomFields.bind( this );
-
-		this.state = { isChecked };
-	}
-
-	toggleCustomFields() {
-		// Submit a hidden form which triggers the toggle_custom_fields admin action.
-		// This action will toggle the setting and reload the editor with the meta box
-		// assets included on the page.
-		document.getElementById( 'toggle-custom-fields-form' ).submit();
-
-		// Make it look like something happened while the page reloads.
-		this.setState( { isChecked: ! this.props.isChecked } );
-	}
-
-	render() {
-		const { label } = this.props;
-		const { isChecked } = this.state;
+export function CustomFieldsConfirmation( { nextState } ) {
+	const [ isReloading, setIsReloading ] = useState( false );
+
+	return (
+		<>
+			<p className="edit-post-options-modal__custom-fields-confirmation-message">
+				{ __( 'A page reload is required for this change. Make sure your content is saved before reloading.' ) }
+			</p>
+			<Button
+				className="edit-post-options-modal__custom-fields-confirmation-button"
+				isDefault
+				isBusy={ isReloading }
+				disabled={ isReloading }
+				onClick={ () => {
+					setIsReloading( true );
+					document.getElementById( 'toggle-custom-fields-form' ).submit();
+				} }
+			>
+				{ nextState === true ? __( 'Enable & Reload' ) : __( 'Disable & Reload' ) }
```

Comment:
This could be simplified.

```suggestion
				{ nextState ? __( 'Enable & Reload' ) : __( 'Disable & Reload' ) }
```